### PR TITLE
Add charge transaction fee

### PR DESCRIPTION
**Pull request by zjb0807**
Closes: #754

### CODE REVIEW

File: ecosystem-modules/ren/renvm-bridge/src/lib.rs [hunk 1]
```diff
@@ -107,13 +121,20 @@ decl_module! {
...
 			#[compact] amount: Balance,
-			_n_hash: [u8; 32],
+			n_hash: [u8; 32],
 			sig: EcdsaSignature,
 		) {
 			ensure_none(origin)?;
-			Self::do_mint(who, amount, sig)?;
+			Self::do_mint(&who, amount, &sig)?;
+
+			let len = (32 + p_hash.len() + n_hash.len() + sig.0.len()) as u32;
```
Comment (xlc) [inline ecosystem-modules/ren/renvm-bridge/src/lib.rs:132]: this is not accurate as it doesn't include everything in a transaction.
should just make build a dummy mint transaction, measure the length, and hardcode the value here

File: ecosystem-modules/ren/renvm-bridge/src/lib.rs [hunk 2]
```diff
@@ -40,16 +46,24 @@ type EcdsaSignature = ecdsa::Signature;
...
-	type Currency: BasicCurrency<Self::AccountId, Balance = Balance>;
+/// Type alias for currency balance.
+pub type BalanceOf<T> = <<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+pub type NegativeImbalanceOf<T> =
+	<<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::NegativeImbalance;
+
+pub trait Config: frame_system::Config {
+	type Event: From<Event<Self>> + Into<<Self as frame_system::Config>::Event>;
+	type Currency: Currency<Self::AccountId>;
+	type CurrencyAdapter: BasicCurrency<Self::AccountId, Balance = Balance>;
```
Comment (xlc) [inline ecosystem-modules/ren/renvm-bridge/src/lib.rs:57]:
```suggestion
	type BridgedTokenCurrency: BasicCurrency<Self::AccountId, Balance = Balance>;
```

File: ecosystem-modules/ren/renvm-bridge/src/lib.rs [hunk 3]
```diff
@@ -113,7 +127,16 @@ decl_module! {
 			sig: EcdsaSignature,
 		) {
 			ensure_none(origin)?;
-			Self::do_mint(who, amount, sig)?;
+			Self::do_mint(&who, amount, &sig)?;
+
+			// Calculated the transaction length from the unit test.
+			let len: u32 = 166;
```
Comment (xlc) [inline ecosystem-modules/ren/renvm-bridge/src/lib.rs:133]: make this a `const MINT_TX_LENGTH: u32 = 166` and have test check the const

**Review by xlc — APPROVED** [review]

File: ecosystem-modules/ren/renvm-bridge/src/lib.rs (hunk 3 above)
Comment (zjb0807) [reply ecosystem-modules/ren/renvm-bridge/src/lib.rs]: I think the dynamic calculation is better because the amount is compact.

**Pull request merged**